scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -530,7 +530,9 @@
                                                    (do
                                                      (log/info "In compute cluster" name ", pod" pod-name
                                                                "was deleted while it was expected starting")
-                                                     (handle-pod-killed compute-cluster pod-name))
+                                                     (if (some-> synthesized-state :pod-preempted-timestamp)
+                                                       (handle-pod-preemption compute-cluster pod-name)
+                                                       (handle-pod-externally-deleted compute-cluster pod-name)))
```

Comment:
Since we have the same three lines in another part of the machine, should we extract a function, e.g. `handle-pod-missing-unexpectedly`?

Reply:
updated